### ISSUE TITLE
Problem with flashlight

### ISSUE THREAD

**Issue by tkapis** [KozGit/DOOM3-BFG-RIFT#3]
Game is really awesome with this mod only after loading save game flashlight is no longer visible in left hand. It works but only when you are touching left touchpad and then only horizontally. Setting a different flashlight mode has no effect. This happens with both the original game and using the All-In-One texture mod.

**Comment by KozGit** [comment]
I took a look at this, and there is a bug where the current flashlight mode is not being saved with the save game file  If the flashlight works, but only moves horizontally when you turn the player, that means its mounted to the body instead of the head, hands or gun.  I just tested, and I was able to work around the issue by binding a control to the 'Next Flashlight Mode' binding, and toggling through all the flashlight modes until the light appeared in the hand again.  ( The four modes are body head gun hand so you will need to toggle thru them all)  Thanks for pointing out the issue, I've already fixed it in code and will hopefully post an update soon.  When the update is released it will unfortunately break existing saves as it adds new data to the file, so any savegames will need to be deleted. Sorry!

**Comment by tkapis** [comment]
The light was moving when I was touching the left trackpad and I moved my left hand not my body. I changed the flashlight mode to weapon via the console but it had no effect. Now I started a new game and the flashlight was on the gun but when I loaded a save game the same problem of course.

I disabled the heading beam but it sometimes reappears after loading a new level.

Also I tried to bind reload to the right grip. It accepted it but it didn't work.

I will wait for the update to play the game again since I can't later use a save from now.

**Comment by puttis86** [comment]
Yeah I have to restart the game as soon as a new level loads due to the heading beam coming back and my target crosshair gets supersmall and I cant interact with any computers/terminals/airlocks etc. its as the scale gets messed up and im to far from the objects I need to touch for the game to recognize me.

im using flashlight on the weapon
no body, just hands.
no headingbeam
and the the circular targeting thing.

when a new level loads the the small circle with a dot in it is so small its just a dot, and the beam is back and I cant interact with stuff. saving the game and restarting and loading that saves makes it all good again.

**Comment by KozGit** [comment]
Ok, I'll take a look at those as well, shouldnt be hard to clean up.

**Comment by KozGit** [comment]
@puttis86 , are you using the oculus native or openvr version?

**Comment by tkapis** [comment]
I hope that the update is coming soon so that I can play the game again.

**Comment by KozGit** [comment]
I'm working on it now... this time of year is just a little crazy!

**Comment by tkapis** [comment]
Of course, I understand, I'll be patient.

I see that your competitor has implemented slots/pockets in the other mod. That's the only thing missing from this mod in fact, maybe you could think about it.

**Comment by Maxxgold** [comment]
I love your MOD! any idea on a time-frame of when the Flashlight bug will be released? No hurry, just wondering.

**Comment by tkapis** [comment]
Samson, is the OpenVR update coming soon, thanks.

**Comment by KozGit** [comment]
Hi,

Sorry for taking so long to address these issues.

There is a new repository for the mod  - you can download an updated version for OpenVR here:

https://github.com/KozGit/DOOM-3-BFG-VR/releases

I apologize, but the OpenVR version has been upgraded, but not tested, so expect some issues.  PLEASE try running the mod in a clean install first, before installing the texture pack.  This will make it much easier to sort out any issues. 
 
If you have any issues, ( and i expect some ), please open an issue in the new repository and I'll get to work on fixing them, I'm not adding any more features until the bugs are worked out of this.

Thx!
